powermon/commands/command_definition.py

```diff
@@ -0,0 +1,51 @@
+from powermon.protocols import ResponseType
+from powermon.dto.command_definition_dto import CommandDefinitionDTO
+
+class CommandDefinition:
+    def __init__(self, code, description, help_text: str, response_type : ResponseType, 
+                 responses, test_responses: list[bytes], regex: str, type: str):
+        if responses is None:
+            raise ValueError(f"responses cannot be None for command_code: {code}")
+        if test_responses is None:
+            raise ValueError(f"test_responses cannot be None for command_code: {code}")
+        self.code = code
```

Comment:
this is the tricky one to 'name' - i think you mentioned somewhere that 'command' is over used and i agree
not sure 'code' is completely there, but is better than command or name.
originally had a hope of having more generic 'commands' that then mapped to the 'code' or multiple codes, but didnt get past getSettings and getStatus (which dont work reliably across all protocols)
so QPI is a code, but what about QED20230912?
its kinda command_text ....

Reply:
I will rename it in a different PR though, it's a lot of changes.

Reply:
Command_string works